### PR TITLE
Fix preset-all

### PR DESCRIPTION
**Pull request by keszybz**
This makes `systemctl preset-all` boring and predicatable, hopefully fixing #3616. It doesn't change anything for template presets, so I'm not marking #2513 as done.

The general idea is that preset-all and preset should be idempotent and modify existing state as little as possible. In addition, make install followed by preset-all or preset on our units should not introduce any changes, i.e. our installation should match our presets.

### CODE REVIEW

**Comment by keszybz** [comment]
It fails in `boot-smoke` — probably not related.

**Comment by martinpitt** [comment]
@keszybz: right, the occasional hanging `serial-getty@ttyS0.service` has haunted us for a fair while, it's unrelated (but unfortunately has never been understood so far). I retried the test.

File: src/systemctl/systemctl.c [hunk 1]
```diff
@@ -6029,7 +6029,7 @@ static int preset_all(int argc, char *argv[], void *userdata) {
 
         if (install_client_side()) {
                 r = unit_file_preset_all(arg_scope, arg_runtime, arg_root, arg_preset_mode, arg_force, &changes, &n_changes);
-                unit_file_dump_changes(r, "preset", changes, n_changes, arg_quiet);
+                unit_file_dump_changes(r, "preset all", changes, n_changes, arg_quiet);
```
Comment (poettering) [inline src/systemctl/systemctl.c]: is this right? The log message might now show up as "Failed to preset all unit, file foobar.service already exists, …" That looks wrong. As the entries in the changes array are for individual unit files, I think "preset" was actually the right parameter to pass here...

**Comment by poettering** [comment]
looks all good to me, except for the msg fix in the first commit.

**Comment by keszybz** [comment]
I dropped the first commit. There's a slight conflict between making the summary message "Failed to %s:", and the per-units messages "Failed to %s unit:" sound both good. I guess "preset" is a better compromise.

**Comment by poettering** [comment]
ok, looks good to merge then.

**Comment by poettering** [comment]
Ci failure appears unrelated. merging.

**Pull request merged**